File: 1º Semestre/Mat Discreta/PROGRAMAS/1º Remessa/relacao.py
print("Relação de conjuntos A e B\n")

list_A = []
el_A = ""
list_B = []
el_B = ""
op = ""
list_A_B = []

try:
    print("Entre com os valores para o conjunto A (digite X para ir para o próximo): ")
    while el_A != "X" and  el_A != "x":
        el_A = str(input('Opção:'))
        if el_A == "X" and  el_A == "x":
            break
        elif el_A != "X" and  el_A != "x":
            list_A.append(el_A)

    print("Entre com os valores para o conjunto B (digite X para sair): ")
    while el_B != "X" and  el_B != "x":
        el_B = str(input('Opção:'))
        if el_B == "X" and  el_B == "x":
            break
        elif el_B != "X" and  el_B != "x":
            list_B.append(el_B)

    print("\nSelecione o tipo de relação que deseja:\n\t\t1-Igualdade\n\t\t2-Maior\n\t\t3-Menor")
    op = input("Opção: ")
    if op == "1":
        for n in list_A:
            for m in list_B:
                if n == m :
                    list_A_B.append(n)
    elif op == "2":
        for n in list_A:
            for m in list_B:
                if n > m:
                    list_A_B.append(n)
    elif op == "3":
        for n in list_A:
            for m in list_B:
                if n < m:
                    list_A_B.append(n)
    elif op != 1 and op != 2 and op != 3 :
        print("Entre com uma opção válida")

    if op == "1":
        print("\nElemntos iguais nos conjuntos:")
    elif op == "2":
        print("\nElementos do conjunto A maiores aos do conjunto B:")
    elif op == "3":
        print("\nElementos do conjunto A menores aos do conjunto B:")

    for n in list_A_B:
        print(n)

except:
    print()

File: 1º Semestre/Mat Discreta/PROGRAMAS/1º Remessa/relacao/src/relacao/Relacao.java
package relacao;
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

public class Relacao {

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		
		List<String> list_A = new ArrayList<String>();
		List<String> list_B = new ArrayList<String>();
		
		String el_A, el_B;
		int x = 0, a, b;
		
		System.out.printf("Quantos elementos quer no Conjunto A? ");
		a = input.nextInt();
		
		System.out.printf("Quantos elementos quer no Conjunto B? ");
		b = input.nextInt();
		
		String[][] relacao = new String[a][b];
		
		System.out.println("Entre com um valor para o conjunto A: ");
		while(x < a) {
			System.out.printf("Opção: ");
			el_A = input.next();
			list_A.add(el_A);
			relacao[x][0] = el_A; 
			System.out.println(relacao[x][0]);
			x++;
		}
		
		x = 0;
		
		System.out.println("\nEntre com um valor para o conjunto B: ");
		while(x < b) {
			System.out.printf("Opção: ");
			el_B = input.next();
			list_B.add(el_B);
			relacao[0][x] = el_B;
			System.out.println(relacao[0][x]);
			x++;
		}
		
		System.out.println("\nElementos juntos:");
		
//		int c = 0, l = 0;
//		while(l < a) {
//			while(c < b) {
//				System.out.println(relacao[l][c]);
//				c++;
//			}
//			l++;
//		}
		
		System.out.println("Elementos de A:\n");
		
		for(String n: list_A) {
			System.out.printf(n, "\n");
		}
		
		System.out.println("\nElementos de B: \n");
		
		for(String n: list_B) {
			System.out.printf(n, "\n");
		}
		
	}

}
